common/sockets.c

```diff
@@ -52,6 +56,15 @@ rfbBool sock_set_nonblocking(rfbSocket sock, rfbBool non_blocking, void (*log)(c
 
 rfbBool sock_wait_for_connected(int socket, unsigned int timeout_seconds)
 {
+#ifdef LIBVNCSERVER_HAVE_POLL
+  struct pollfd pfd;
+  pfd.fd = socket;
+  pfd.events = POLLIN | POLLPRI;
+
+  if (poll(&pfd, 1, timeout_seconds*1000)==1) {
+    if ((pfd.revents & POLLERR) || (pfd.revents & POLLHUP))
```

Comment:
AFAIK poll.h and `poll()` don't exist at all on Win32 so there's nothing to worry here about since `LIBVNCSERVER_HAVE_POLL{_H}` won't be set anyway. WinSock2 has `WSAPoll()`, so an alternative code path could be added later.

Reply:
Well they have https://docs.microsoft.com/en-us/windows/win32/api/winsock2/nf-winsock2-wsapoll, but that probably complicated matters even further...